spiketools/tests/spatial/test_occupancy.py

```diff
@@ -73,65 +65,139 @@ def test_compute_bin_assignment():
-    bin_firing = np.array([[0, 1, 0], [1, 2, 0]])
+    bin_counts = np.array([[0, 1, 0], [1, 2, 0]])
     occupancy = np.array([[0, 2, 1], [1, 1, 0]])
-    normed_bf = normalize_bin_firing(bin_firing, occupancy)
-    assert isinstance(normed_bf, np.ndarray)
+    normed_counts = normalize_bin_counts(bin_counts, occupancy)
+    assert isinstance(normed_counts, np.ndarray)
     expected = np.array([[np.nan, 0.5, 0.], [1., 2., np.nan]])
-    assert np.array_equal(normed_bf, expected, equal_nan=True)
+    assert np.array_equal(normed_counts, expected, equal_nan=True)
+
+def test_create_position_df():
+
+    timestamps = np.array([5, 5, 5, 5])
+
+    # 1d case
+    bins = 2
+    position = np.array([1, 2, 4, 5])
+    df = create_position_df(position, timestamps, bins)
+    assert isinstance(df, pd.DataFrame)
+    assert len(df) == position.shape[-1]
+    assert np.array_equal(df.xbin.values, np.array([0, 0, 1, 1]))
+
+    # check speed dropping
+    speed = np.array([1, 1, 0, 1])
+    df = create_position_df(position, timestamps, bins, speed=speed, speed_threshold=0.5)
+    assert len(df) == sum(speed)
+    assert np.array_equal(df.xbin.values, np.array([0, 0, 1]))
+
+    # 2d case
+    bins = [2, 2]
+    position = np.array([[1, 2, 4, 5], [5, 4, 2, 1]])
+    df = create_position_df(position, timestamps, bins)
+    assert isinstance(df, pd.DataFrame)
+    assert len(df) == position.shape[-1]
+    assert np.array_equal(df.xbin.values, np.array([0, 0, 1, 1]))
+    assert np.array_equal(df.ybin.values, np.array([1, 1, 0, 0]))
 
-def test_compute_bin_time():
+def test_compute_occupancy_df():
 
-    # define a timestamp, with irregular times
-    timestamp = np.array([0.0, 1.0, 2.0, 3.0, 4.5, 5.0, 6.0, 7.0, 8.0, 9.0, 12.0])
-    bin_time = compute_bin_time(timestamp)
+    data_dict = {
+        'time' : np.array([5, 5, 5, 5, 5, 5, 5, 5, 5.]),
+        'xbin' : np.array([0, 0, 0, 0, 1, 1, 1, 2, 2])}
 
-    # check dimensions & sum
-    assert bin_time.shape[0] == timestamp.shape[0]
-    assert np.sum(np.diff(timestamp)) == np.sum(bin_time)
+    # check 1d case
+    bins = 3
+    bindf = pd.DataFrame(data_dict)
+    occ = compute_occupancy_df(bindf, bins)
+    assert isinstance(occ, np.ndarray)
+    assert np.array_equal(occ, np.array([20, 15, 10]))
+
+    # check 2d case
+    data_dict['ybin'] = np.array([0, 0, 1, 1, 0, 1, 1, 0, 1])
+    bins = [3, 2]
+    bindf = pd.DataFrame(data_dict)
+    occ = compute_occupancy_df(bindf, bins)
+    assert isinstance(occ, np.ndarray)
+    assert np.array_equal(occ.shape, np.array([bins[1], bins[0]]))
+    assert np.array_equal(occ, np.array([[10, 5, 5], [10, 10, 5]]))
+
+    # check minimum
+    occ = compute_occupancy_df(bindf, bins, minimum=6)
+    assert np.array_equal(occ, np.array([[10, 0, 0], [10, 10, 0]]))
+
+    # check normalization
+    occ = compute_occupancy_df(bindf, bins, normalize=True)
+    assert np.sum(occ) == 1.0
+
+    # check nans
+    data_dict['time'] = np.array([5, 5, 5, 5, 5, 5, 5, 0, 0.])
```

Comment:
Very minor - extra point after the last zero?

Reply:
since `time` should be a float, adding the point makes that value (and thus the whole array) a float array instead of an integer array. I don't remember if it's important to do that here hahahah, but I'm guessing that's why!

For aesthetic simplicity, I update to have all entries be float (though that's not really needed).